test/geometry/liegroup/test_so2.py

```diff
@@ -11,15 +11,17 @@ def _make_rand_data(self, device, dtype, input_shape):
         shape = input_shape[1:] if batch_size is None else input_shape
         return torch.rand(shape, device=device, dtype=dtype)
 
-    def test_smoke(self, device, dtype):
-        z = torch.randn(2, 1, dtype=torch.cfloat, device=device)
+    @pytest.mark.parametrize("cdtype", (torch.cfloat, torch.cdouble))
+    def test_smoke(self, device, cdtype):
+        z = torch.randn(2, 1, dtype=cdtype, device=device)
         s = So2(z)
         assert isinstance(s, So2)
         self.assert_close(s.z.data, z.data)
 
     @pytest.mark.parametrize("input_shape", [(1,), (2,), (5,), ()])
-    def test_cardinality(self, device, dtype, input_shape):
-        z = torch.randn(input_shape, dtype=torch.cfloat, device=device)
+    @pytest.mark.parametrize("cdtype", (torch.cfloat, torch.cdouble))
```

Comment:
i think this won't work since it will generate imcompatible combinations  like torch.float32/torch.cfloat64

Reply:
For this particular test, it shouldn't give out these errors since it only checks the shapes. I did run into some issues when testing this out on __mul__, any other way we should inject complex type data?

Reply:
oki